homeassistant/components/home_connect/entity.py

```diff
@@ -121,3 +128,34 @@ async def async_set_option(self, value: str | float | bool) -> None:
     def bsh_key(self) -> OptionKey:
         """Return the BSH key."""
         return cast(OptionKey, self.entity_description.key)
+
+
+def constraint_fetcher[_EntityT: HomeConnectEntity, **_P](
+    func: Callable[Concatenate[_EntityT, _P], Coroutine[Any, Any, Any]],
+) -> Callable[Concatenate[_EntityT, _P], Coroutine[Any, Any, None]]:
+    """Decorate the function to catch Home Connect too many requests error and retry later.
+
+    If it needs to be called later, it will call async_write_ha_state function
+    """
+
+    async def handler_to_return(
+        self: _EntityT, *args: _P.args, **kwargs: _P.kwargs
+    ) -> None:
+        async def handler(_datetime: datetime | None = None) -> None:
+            try:
+                await func(self, *args, **kwargs)
+            except TooManyRequestsError as err:
+                if (retry_after := err.retry_after) is None:
+                    retry_after = API_DEFAULT_RETRY_AFTER
```

Comment:
I ran the tests locally and we're not covering the default retry after time case here on line 149. It would be good to add a test for that too. We can patch the constant to 0 with:

```py
with patch("homeassistant.components.home_connect.entity.API_DEFAULT_RETRY_AFTER", new=0):
```